src/app/attendance/components/AttendanceFormBuilder.tsx

```diff
@@ -0,0 +1,270 @@
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import {useMemo } from "react";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import {
+  Select,
+  SelectTrigger,
+  SelectContent,
+  SelectItem,
+  SelectValue,
+} from "@/components/ui/select";
+import {
+  Form,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormControl,
+  FormMessage,
+} from "@/components/ui/form";
+import { User } from "@/types";
+import { MemberRole, UserStatus } from "@/lib/routes";
+
+const attendanceSchema = z
+  .object({
+    memberId: z.number({
+      required_error: "Please select user name",
+    }),
+    startTime: z.string().min(1, "Start time is required"),
+    endTime: z.string().min(1, "End time is required"),
+  })
+  .refine(
+    (data) => {
+      const startTime = new Date(`1970-01-01T${data.startTime}:00`);
+      let endTime = new Date(`1970-01-01T${data.endTime}:00`);
+      if (endTime < startTime) endTime.setDate(endTime.getDate() + 1);
+      const nextTwoHours = new Date(startTime.getTime() + 2 * 60 * 60 * 1000);
+      return endTime <= nextTwoHours;
+    },
+    {
+      message: "End time must be later then start time",
+      path: ["endTime"],
+    }
+  );
+
+type AttendanceFormValues = z.infer<typeof attendanceSchema>;
+
+const loginUser: User = {
+  id: 1,
+  name: "Alice Johnson",
+  shift: "A",
+  role: "incharge",
+  status: "active",
+  phone: "123-456-7890",
+  address: "123 Main St, Anytown, USA",
+  created_at: new Date().toISOString(),
+  updated_at: new Date().toISOString(),
+  deleted_at: null,
+};
+
+const members = [
+  { id: 123, name: "John Doe", role: "member", shift: "A", status: "active" },
+  {
+    id: 1,
+    name: "Alice Johnson",
+    shift: "A",
+    role: "incharge",
+    status: "active",
+  },
+  { id: 2, name: "Bob Smith", shift: "B", role: "member", status: "invited" },
+  {
+    id: 3,
+    name: "Charlie Brown",
+    shift: "C",
+    role: "shift-incharge",
+    status: "active",
+  },
+  { id: 4, name: "Diana Prince", shift: "D", role: "member", status: "active" },
+  {
+    id: 5,
+    name: "Ethan Hunt",
+    shift: "A",
+    role: "incharge",
+    status: "invited",
+  },
+  {
+    id: 6,
+    name: "Fiona Gallagher",
+    shift: "B",
+    role: "member",
+    status: "active",
+  },
+  {
+    id: 7,
+    name: "George Banks",
+    shift: "B",
+    role: "shift-incharge",
+    status: "active",
+  },
+  {
+    id: 8,
+    name: "Hannah Baker",
+    shift: "B",
+    role: "member",
+    status: "invited",
+  },
+  {
+    id: 9,
+    name: "Ian Malcolm",
+    shift: "C",
+    role: "incharge",
+    status: "active",
+  },
+  {
+    id: 10,
+    name: "Jack Sparrow",
+    shift: "C",
+    role: "member",
+    status: "active",
+  },
+  {
+    id: 11,
+    name: "Kara Danvers",
+    shift: "D",
+    role: "shift-incharge",
+    status: "invited",
+  },
+];
+
+const AttendanceFormBuilder = () => {
```

Comment:
Pass on members and user as prop.
@muhammad-bilal-ali-khokhar create a side task to get user information from session.

Reply:
```suggestion
const AttendanceFormBuilder = ({members = members, user = loginUser }) => {
```